client/blocks/inline-help/inline-help-search-results.jsx

```diff
@@ -15,60 +16,145 @@ import { recordTracksEvent } from 'state/analytics/actions';
 import QueryInlineHelpSearch from 'components/data/query-inline-help-search';
 import PlaceholderLines from './placeholder-lines';
 import { decodeEntities, preventWidows } from 'lib/formatting';
-import getInlineHelpSearchResultsForQuery from 'state/inline-help/selectors/get-inline-help-search-results-for-query';
+import getSearchResultsByQuery from 'state/inline-help/selectors/get-inline-help-search-results-for-query';
 import getSelectedResultIndex from 'state/inline-help/selectors/get-selected-result-index';
+import getInlineHelpCurrentlySelectedResult from 'state/inline-help/selectors/get-inline-help-currently-selected-result';
 import isRequestingInlineHelpSearchResultsForQuery from 'state/inline-help/selectors/is-requesting-inline-help-search-results-for-query';
 import hasInlineHelpAPIResults from 'state/selectors/has-inline-help-api-results';
 import { selectResult } from 'state/inline-help/actions';
 import { localizeUrl } from 'lib/i18n-utils';
+import Gridicon from 'components/gridicon';
+import {
+	SUPPORT_TYPE_ADMIN_SECTION,
+	SUPPORT_TYPE_API_HELP,
+	SUPPORT_TYPE_CONTEXTUAL_HELP,
+} from './constants';
 
 function HelpSearchResults( {
 	hasAPIResults = false,
 	isSearching = false,
-	openResult,
+	onSelect,
 	searchQuery = '',
-	searchResults,
+	searchResults = [],
+	selectedResult = {},
 	selectedResultIndex = -1,
 	selectSearchResult,
 	translate = identity,
 	placeholderLines,
+	track,
 } ) {
-	const selectResultHandler = ( selectionIndex ) => ( event ) => {
-		const selectedResult = searchResults?.[ selectionIndex ] ?? null;
-		selectSearchResult( selectionIndex );
-		openResult( event, selectedResult );
+	const supportTypeRef = useRef( searchResults?.[ 0 ]?.support_type );
+
+	function getTitleBySectionType( addSection, type, query = '' ) {
+		if ( ! addSection ) {
+			return null;
+		}
+
+		let title = '';
+		switch ( type ) {
+			case SUPPORT_TYPE_CONTEXTUAL_HELP:
+				if ( ! query.length ) {
+					return;
```

Comment:
We `return` here but `return null` in default. Is that intentional?

Reply:
No, it isn't. It should be `null` since it's a react component. Changing on this PR. Good catch 👍

Reply:
Tackled here 008a42a93e23dee8b4754f5dcf8c4e767c0db3d8. Thanks!